### ISSUE TITLE
Remote Control Learning not working for Debian 12 (MX Linux), liblo7 version 0.31-1

### ISSUE THREAD

**Issue by Stazed** [Stazed/non-mixer-xt#32]
@zigmhount I'm moving this issue here since the original was simply a notification.
I cannot duplicate this issue, on Manjaro, Ubuntu Studio, or Fedora. It seems specific to your machine, or perhaps Debian 12.

Can you confirm that reverting the last nonlib commit fixes the problem?

Also, it would help if you could get the mixer terminal messages. I could not find any way to get Raysession to print client messages. I use new-session-manager for debugging NSM sessions, and run it in a terminal. Maybe build with debugging enabled and check for any warning message from midi-mapper-xt or non-mixer-xt.

I just pushed a commit (wip branch) for the menu parsing problem but that should not have any impact on the remote control learning.

**Comment by zigmhount** [comment]
Well... I re-built using the previous commit of nonlib and I could successfully map MIDI controls.

But then I re-built using the latest commit of nonlib and it does still work 🤦🏼  so I'm closing this issue. Very sorry for the noise !

---

I'm no Git expert, so I don't really understand what happened, but it seems that maybe `git submodule update` may not be enough to pull the changes in the submodules?

Here is what I did, maybe you can help so I do not make the same mistake again?
- Before reverting nonlib to the previous commit:
```
> non-mixer-xt/nonlib$ git status
HEAD detached at f8dbfe3
nothing to commit, working tree clean
> non-mixer-xt/nonlib$ git checkout 7c19a105e1ff1c23fb115084fc71e184c6f37485
Previous HEAD position was f8dbfe3 Fix for LibLO 0.32 which now requires leading '/' for path.
HEAD is now at 7c19a10 Reset log_start(), log_end() to public for nudging.
```
- built and installed non-mixer-xt and tested
- reverted to the latest commit of nonlib:
```
> non-mixer-xt/nonlib$ git status
HEAD detached at 7c19a10
nothing to commit, working tree clean
> non-mixer-xt/nonlib$ git checkout main
Previous HEAD position was 7c19a10 Reset log_start(), log_end() to public for nudging.
Switched to branch 'main'
Your branch is behind 'origin/main' by 5 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
> non-mixer-xt/nonlib$ git status
On branch main
Your branch is behind 'origin/main' by 5 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)

nothing to commit, working tree clean
> non-mixer-xt/nonlib$ git pull
Updating d949d5f..f8dbfe3
Fast-forward
 Loggable.C     | 1 +
 Loggable.H     | 4 ++--
 OSC/Endpoint.C | 2 +-
 3 files changed, 4 insertions(+), 3 deletions(-)
> non-mixer-xt/nonlib$ git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
```
- built again and tested and it works.

If I had already properly checked out the latest commit f8dbfe3, I should not have had to pull again after checking out main, should I? 

Anyway, next time I'll double-check by pulling the submodules manually before raising a bug report.

**Comment by Stazed** [comment]
It actually looks like your submodule status was correct and everything was working properly. I suspect that this is one of those rare cases I have seen in which make does not recognize that the submodule file was changed, perhaps for the midi-mapper-xt build. Sometimes  after an update, deleting the build directory is best, or running make clean first...
Just guessing.  Anyway, good to know that it is working now. 

And you did find the broken parsing for the connect/disconnect menu so still a useful bug report!!!
Thanks.
